rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -731,6 +731,9 @@ public MavenRepository normalizeRepository(MavenRepository originalRepository, M
                 String httpsUri = repository.getUri().toLowerCase().startsWith("http:") ?
                         repository.getUri().replaceFirst("[hH][tT][tT][pP]://", "https://") :
                         repository.getUri();
+                if (!httpsUri.endsWith("/")) {
+                    httpsUri += "/";
```

Comment:
Based on [this comment](https://github.com/openrewrite/rewrite-maven-plugin/issues/742#issuecomment-1961240404).